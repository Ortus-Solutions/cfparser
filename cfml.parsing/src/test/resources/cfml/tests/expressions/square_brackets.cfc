component{
function foo(){
var orderedStruct = [ keyA: 'value', keyB: 'value' ];
}
}