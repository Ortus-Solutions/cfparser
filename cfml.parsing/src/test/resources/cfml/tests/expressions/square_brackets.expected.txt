/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
Hidden:NEWLINE                  <>
FUNCTION                 <function>
IDENTIFIER               <foo>
'('                      <(>
')'                      <)>
'{'                      <{>
Hidden:NEWLINE                  <>
VAR                      <var>
IDENTIFIER               <orderedStruct>
'='                      <=>
'['                      <[>
IDENTIFIER               <keyA>
':'                      <:>
OPEN_STRING              <'>
STRING_LITERAL           <value>
CLOSE_STRING             <'>
','                      <,>
IDENTIFIER               <keyB>
':'                      <:>
OPEN_STRING              <'>
STRING_LITERAL           <value>
CLOSE_STRING             <'>
']'                      <]>
';'                      <;>
Hidden:NEWLINE                  <>
'}'                      <}>
Hidden:NEWLINE                  <>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          function
          (identifier foo)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (localAssignmentExpression
                var
                (startExpression
                  (baseExpression
                    (unaryExpression (memberExpression (identifier orderedStruct)))
                  )
                )
                =
                (startExpression
                  (baseExpression
                    (unaryExpression
                      (primaryExpression
                        (implicitOrderedStruct
                          [
                          (implicitStructElements
                            (implicitStructExpression
                              (implicitStructKeyExpression (multipartIdentifier (identifier keyA)))
                              :
                              (baseExpression
                                (unaryExpression
                                  (primaryExpression
                                    (literalExpression (stringLiteral ' (stringLiteralPart value) '))
                                  )
                                )
                              )
                            )
                            ,
                            (implicitStructExpression
                              (implicitStructKeyExpression (multipartIdentifier (identifier keyB)))
                              :
                              (baseExpression
                                (unaryExpression
                                  (primaryExpression
                                    (literalExpression (stringLiteral ' (stringLiteralPart value) '))
                                  )
                                )
                              )
                            )
                          )
                          ]
                        )
                      )
                    )
                  )
                )
              )
              (endOfStatement ;)
            )
            }
          )
        )
      )
      }
    )
  )
)
/*======*/
/*===DECOMPILE===*/
component {
  public function foo()     {
      var orderedStruct = [keyA:'value',keyB:'value'];

    }
}
/*======*/